### PR TITLE
feat(sink): use mysql and postgresql container in jdbc unit test

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Previously, we didn't run the java unit test in our CI. If we run it, we will fail in the jdbc sink test. It fails because we use `derby` in our unit test, while we cannot get the primary key of `derby` via jdbc, which leads to some failures.

In this PR, we switch to use the test container of both MySQL and PostgreSQL so that we don't suffer from `derby` any more. And we enable the java unit test in CI by the way.

Besides, in this PR, we added a logic to report errors when the handling update and delete event for downstream table without a primary key.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by WillyKidd — COMMENTED** [review]
ohhh...PostgresSourceTest failed because the tests were run in the following order:
1. testPermissionCheck
2. testLines

**Review by WillyKidd — COMMENTED** [review]
oooops...`PostgresSourceTest` failed because the tests could be randomly run in the following order:
1. testPermissionCheck
2. testLines

interestingly, the `testPermissionCheck` case will revoke some permissions from the user `Postgres`,   thus failing the `testLines` check

let me see how to fix it..

**Review by StrikeW — COMMENTED** [review]
LGTM, if unit test workflow pass.

File: java/connector-node/risingwave-sink-jdbc/pom.xml [hunk 1]
```diff
@@ -52,8 +56,21 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.derby</groupId>
-            <artifactId>derby</artifactId>
+            <groupId>org.testcontainers</groupId>
```
Comment (StrikeW) [inline java/connector-node/risingwave-sink-jdbc/pom.xml:59]: May declare these dependencies in root pom and reference here?

**Review by StrikeW — APPROVED** [review]
Rest LGTM

**Review by tabVersion — APPROVED** [review]

File: java/connector-node/risingwave-sink-jdbc/pom.xml (hunk 1 above)
Comment (wenym1) [reply java/connector-node/risingwave-sink-jdbc/pom.xml:59]: For the test dependency I think we can declare them here directly.

**Pull request merged**